### DISCUSSION TITLE
How to extend PATH to include ~/.local/bin for WSL?

### DISCUSSION THREAD

**Discussion by ssbarnea** [Vampire/setup-wsl#22] [Q&A] [answered]
One issue for which I am still trying to find an answer is how do I make this common pattern work in under WSL, for further `run:` steps.

```
if ! [[ "$PATH" =~ "$HOME/.local/bin" ]]; then
then
    PATH="$HOME/.local/bin:$PATH"
fi
export PATH
```

The example above is a common pattern for adding the user bin directory to user PATH, and often that bit is appended to `.bashrc` so further scripts executed would make use of it.

The problem is that wsl bash session is not loading the `.bashrc` file, so I am unable to make the PATH change persistent.

**Comment by briantist** [comment]
Hey @ssbarnea , I was researching this and expected it to be something where you would modify the custom shell command via this action, so that you could change the invocation to add `-i` or something, but it looks like there's a much better way it could be done directly in the workflow via `WSLENV`, as described here:
https://github.com/Vampire/setup-wsl/discussions/9#discussioncomment-993367

**Reply by ssbarnea** [reply]
`PATH` is not something you can really set (or want) using the workflow itself, especially when using a test matrix that spawns across windows, macos and linux.

Based on my tests with WSL1 on GHA, the .bashrc file was not loaded.

**Reply by Vampire** [reply, marked as answer]
I agree to both of what Brian said.

Either you change the `wsl-shell-command` input to something that will evaluate the `.bashrc` file.
Adding `-i` should not be sufficient as the default has `--norc`.
Leaving out `--norc` should probably be enough, or if not then maybe additionally `-i`.

Or as Brian said, using the environment variable support.
